### PR TITLE
Template for Terminal Challenge

### LINKED ISSUE #67: Add tab layout on mobile to Chapter 1, challenge 1, part 2 page
[The page in question](https://savingsatoshi.com/chapters/chapter-1/genesis/genesis-pt2).

[Chapter 1, challenge 1, part 1](https://savingsatoshi.com/chapters/chapter-1/genesis) ([code](https://github.com/saving-satoshi/saving-satoshi/blob/master/app/chapters/%5Bslug%5D/genesis/page.tsx)) is a good reference for how to implement this.

- Show tabs on mobile to switch between `info` and `code`
- Separate tab content out from main page lass into separate components
- Various mobile layout tweaks in the page and in components (let me know if I need to list those - would be great if we could templatize this somehow)

**Screenshot:**

image

**Reference design for Terminal challenge pages ([Figma](https://www.figma.com/file/LqjK3Tpvd9KJ4buFArCJBQ/Saving-Satoshi?node-id=1056%3A14226&t=nxCE26gS2fD24CVh-1)):**

image

### LINKED ISSUE #68: Add tab layout on mobile to Chapter 1, challenge 2, part 2 page
This is essentially the same task as #67, but for [another page](https://savingsatoshi.com/chapters/chapter-1/transacting/transacting-2) based on the same template. 

Screenshot:
image

### PR DESCRIPTION
**Pull request by vinayaksh42**
Fixes #67, Fixes #68 

- Created a Template for Terminal-based Challenges
- Implemented the template for Chapter 1, challenge 1, part 2 and Chapter 1, challenge 2, part 2
- Props required by the template 
  - `expectedInput` : answer to the challenge problem
  - `saveInfo` : information required for saving user progress
  - `next` : link to next part of chapter
  - `instruction` : terminal instruction for user
  - `successMessage` : Message displayed to the user upon finishing a challenge

Screenshots:
Screenshot 2023-02-02 at 6 33 32 PM
Screenshot 2023-02-02 at 6 33 17 PM

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,6 +12,7 @@
 # next.js
 /.next/
 /out/
+/.contentlayer/
```
Comment (afterburn) [inline .gitignore:15]: We're not using ContentLayer anymore so I think we don't need this. You might still have a contentlayer folder on your machine from before the architecture refactor, but you can remove it.

**Comment by afterburn** [comment]
Nice! Left a small nitpicky comment and it's not in the design yet, but I think we could use some spacing here: 
image

File: .gitignore (hunk 1 above)
Comment (vinayaksh42) [reply .gitignore]: okay, thanks for the info. removed `.contentLayer` from `.gitignore`

**Comment by vinayaksh42** [comment]
> Nice! Left a small nitpicky comment and it's not in the design yet, but I think we could use some spacing here:
> image

In the Figma mockup we have a 30px space between `tips` and `copyable code`, I have added a 30pixel margin

**Review by afterburn — APPROVED** [review]

**Review by ecurrencyhodler — APPROVED** [review]

**Pull request merged**